vector/src/debug/java/im/vector/app/leakcanary/LeakCanaryProxy.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.leakcanary
+
+import leakcanary.LeakCanary
+import javax.inject.Inject
+
+class LeakCanaryProxy @Inject constructor() {
```

Comment:
it hasn't been merged yet but we'll be favouring variant injection rather than file replacement with https://github.com/vector-im/element-android/pull/6783/files

which means having a `LeakDetector` interface with `NoopLeakDetector` and `LeakCanaryLeakDetector` implementations provided by debug and release

this allows a top level `:vector-app` module to provide the implementations (or even extracted to separate modules)

Reply:
Thanks for the suggestion. I have done an update and rebased my branch on yours.